content/docs/user-guide/dvc-files-and-directories.md

```diff
@@ -138,6 +162,45 @@ the possible following fields:
 
 `dvc.yaml` files also support `# comments`.
 
+### dvc.lock file
+
+For every `dvc.yaml` file, a corresponding `dvc.lock` (YAML) file is created or
+updated by DVC as well. Its purpose is to lock the exact contents of all files
+involved in the pipeline(s) at the time that the pipeline was last committed
+(after using `dvc add`, `dvc run`, `dvc repro`, or `dvc commit`, for example).
+
+This means that regular <abbr>dependencies</abbr> and all <abbr>outputs</abbr>
+(including [metrics](/doc/command-reference/metrics) and
+[plots](/doc/command-reference/plots) files) are also listed in `dvc.lock`, but
+with additional fields that store the `md5` hash value (or `etag` for HTTP
+<abbr>external dependencies</abbr>) of each file or directory tracked by DVC.
+
+[parameter](/doc/command-reference/params#examples) key/value pairs are listed
```

Comment:
not sure we need that verbose descritpion

Reply:
OK. Can you PTAL now? First there's a description of what dvc.lock is for, then the example, and at the bottom, these verbose descriptions of what's in it: stage commands, deps and outs and a bullet list of kinds of file hashes, and params key/value pairs.

Reply:
Is all the bottom text after the example unnecessary in your opinion?